provider/sidecar.go

```diff
@@ -266,18 +266,27 @@ func (provider *Sidecar) makeBackends(sidecarStates *catalog.ServicesState) map[
 			}
 
 			if svc.IsAlive() {
-				for i := 0; i < len(svc.Ports); i++ {
-					ipAddr, err := net.LookupIP(svc.Hostname)
-					if err != nil {
-						log.Errorln("Error resolving Ip address, ", err)
-						backend.Servers[svc.Hostname] = types.Server{
-							URL: "http://" + svc.Hostname + ":" + strconv.FormatInt(svc.Ports[i].Port, 10),
-						}
-					} else {
-						backend.Servers[svc.Hostname] = types.Server{
-							URL: "http://" + ipAddr[0].String() + ":" + strconv.FormatInt(svc.Ports[i].Port, 10),
+				for _, port := range svc.Ports {
+					name := svc.Hostname
+					if len(svc.Ports) > 1 {
+						name = fmt.Sprintf("%s_%d", svc.Hostname, port.Port)
+					}
+
+					host := port.IP
+					if host == "" {
```

Comment:
I don't do this in Sidecar and should, but let's be careful here to make sure we got a properly formed IP address by doing an [`net.ParseIP`](https://golang.org/pkg/net/#ParseIP) here to make sure it's a valid IP as well.